Sources/App/Controllers/PackageController+routes.swift

```diff
@@ -227,11 +227,23 @@ enum PackageController {
             .init(archive: $0, isCurrent: $0.name == archive)
         }
 
-        let canonicalUrl: String? = {
-            guard let canonicalTarget = documentationMetadata.canonicalTarget else { return nil }
+        let canonicalUrl: String? = try await {
+            // The *only* place to get the canonical capitalisation is the database.
+            let repository = try await Repository.query(on: req.db)
+                .field(\.$owner)
+                .field(\.$name)
+                .filter(Repository.self, \.$owner, .custom("ilike"), owner)
+                .filter(Repository.self, \.$name, .custom("ilike"), repository)
+                .first()
+            
+            guard let repositoryName = repository?.name,
+                  let repositoryOwner = repository?.owner,
+                  let canonicalTarget = documentationMetadata.canonicalTarget
+            else { return nil }
```

Comment:
Could we not simply make a `lowercased()` owner/repo pair the canonical URL? I'd hate to have to run an extra query on all these requests just for that lookup.

Reply:
Alternatively, we should pull the fields from the upstream call to `DocumentationMetadata.query` and pass them through instead of the query parameter values.

Reply:
(Sorry, just realised this was still a draft)